File: elixir/bob/bob.exs
defmodule Bob do
  def hey(input) do
    cond do
      String.trim(input) == "" -> "Fine. Be that way!"
      String.ends_with?(input, "?") -> "Sure."
      String.match?(input, ~r/[A-Z]/) and
        String.upcase(input) == input -> "Whoa, chill out!"
      !String.match?(input, ~r/[a-zA-Z0-9\s]/) -> "Whoa, chill out!"
      true -> "Whatever."
    end
  end
end

File: clojure/bob/src/bob.clj
(ns bob
  (:require [clojure.string :refer [upper-case ends-with? trim lower-case]]
))

(defn response-for
  [input]
  (cond
    (= "" (trim input)) "Fine. Be that way!"
    (and
      (= input (upper-case input))
      (not (= input (lower-case input)))) "Whoa, chill out!"
    (ends-with? input "?") "Sure."
    true "Whatever."
  )
)
